### PR TITLE
Add mutations responsible for managing the discount assigned to order

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds a new mutation responsible for handling the discounts for draft orders.

**FYI. ALL FIXME comments will be handled in separate PR**

https://app.clickup.com/t/2549495/SALEOR-1877
https://app.clickup.com/t/2549495/SALEOR-1876

# Impact

* [ ] New migrations
* [X] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1877 Add discount mutations](https://app.clickup.com/t/2549495/SALEOR-1877)

File: saleor/order/utils.py [hunk 1]
```diff
@@ -131,37 +148,37 @@ def recalculate_order_weight(order):
...
+
+
+def update_order_prices(order):
     """Update prices in order with given discounts and proper taxes."""
     manager = get_plugins_manager()
-    channel = order.channel
-    for line in order:  # type: OrderLine
+
+    lines_to_update = []
+    for line in order.lines.all():  # type: OrderLine
```
Comment (fowczarek) [inline saleor/order/utils.py:167]: Should we prefetch products here?

File: saleor/graphql/order/types.py [hunk 2]
```diff
@@ -499,20 +550,24 @@ class Meta:
...
 
     @staticmethod
-    def resolve_discount(root: models.Order, _info):
-        discount = root.discounts.first()
+    def resolve_discounts(root: models.Order, info):
+        return OrderDiscountsByOrderIDLoader(info.context).load(root.id)
+
+    @staticmethod
+    def resolve_discount(root: models.Order, info):
+        discount = root.discounts.filter(type=OrderDiscountType.VOUCHER).first()
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:558]: Could we use dataloader here?

File: saleor/graphql/order/types.py [hunk 3]
```diff
@@ -499,20 +550,24 @@ class Meta:
...
         return (
             Money(amount=discount.value, currency=root.currency) if discount else None
         )
 
     @staticmethod
-    def resolve_discount_name(root: models.Order, _info):
-        discount = root.discounts.first()
+    def resolve_discount_name(root: models.Order, info):
+        discount = root.discounts.filter(type=OrderDiscountType.VOUCHER).first()
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:565]: Could we use dataloader here?

File: saleor/graphql/order/types.py [hunk 4]
```diff
@@ -499,20 +550,24 @@ class Meta:
...
         return discount.name if discount else None
 
     @staticmethod
-    def resolve_translated_discount_name(root: models.Order, _info):
-        discount = root.discounts.first()
+    def resolve_translated_discount_name(root: models.Order, info):
+        discount = root.discounts.filter(type=OrderDiscountType.VOUCHER).first()
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:570]: Could we use dataloader here?

File: saleor/order/utils.py [hunk 5]
```diff
@@ -131,37 +148,37 @@ def recalculate_order_weight(order):
...
+
+    price = manager.calculate_order_line_unit(order, line, variant, product)
+    if price != line.unit_price:
+        line.unit_price = price
+        line.total_price = line.unit_price * line.quantity
+        if price.tax and price.net:
+            line.tax_rate = manager.get_order_line_tax_rate(order, product, None, price)
+
+
+def update_order_prices(order):
```
Comment (fowczarek) [inline saleor/order/utils.py:162]: Maybe we should pass the plugin manager in as many places as possible? Add add
```
if not manager:
    manager = get_plugins_manager()
```

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/order/types.py [hunk 6]
```diff
@@ -468,6 +515,12 @@ class Order(CountableDjangoObjectType):
         ),
     )
 
+    discounts = graphene.List(
```
Comment (fowczarek) [inline saleor/graphql/order/types.py:518]: Could we add it to the existing benchmark or create a new benchmark for getting few orders with many discounts?

**Review by fowczarek — APPROVED** [review]

**Pull request merged**